python/pyproject.toml

```diff
@@ -17,7 +17,7 @@
 
 [tool.poetry]
 name = "pyiceberg"
-version = "0.0.1rc1"
+version = "0.14.0.dev0"
```

Comment:
I guess using 0.14.0 means every time there's major/minor release from java library, there's need to bump up this to keep it in sync. However I feel there's still cognitive benefit so LGTM!

Reply:
@Fokko how would this change for patch releases to the python client?

Reply:
I'd probably set this to `0.1.0.dev0`? I don't think that we want to tie Java and Python versions together, right?

Reply:
As I wrote in the initial post, this is up for discussion. I can see benefits on both sides.

The direct reason for adding the version to the Python code is to pass it as a header to the REST API. If we release Python and Java together, we can give some guarantee that they are compatible. 
My biggest concern with tying them together is that the release process becomes more involved because we have to test both Java and Python. Especially since the Python codebase is still young, I would expect a faster release cadence until everything settles.

I've changed it to `0.1.0.dev0` for now.